zip-0320.rst

```diff
@@ -81,50 +80,78 @@ Requirements
 following steps:
 
 1. Decode the address to a byte sequence using the Base58Check decoding
    algorithm [#Base58Check]_.
 2. If the length of the resulting byte sequence is not 22 bytes or if its two-byte
-   address prefix is not :math:`[\mathtt{0x1C}, \mathtt{0xB8}]`, return an error. Otherwise,
-   let the **validating key hash** be the remaining 20 bytes of the sequence after
-   removing the two-byte address prefix.
+   address prefix is not :math:`[\mathtt{0x1C}, \mathtt{0xB8}]`, return an error.
+   Otherwise, let the **validating key hash** be the remaining 20 bytes of the
+   sequence after removing the two-byte address prefix.
 3. Reencode the 20-byte **validating key hash** using the Bech32m encoding
    defined in [#bip-0350]_ with the human-readable prefix (HRP) ``"tex"``.
 
 For Testnet addresses, the required lead bytes of a P2PKH address in step 2 are
 :math:`[\mathtt{0x1D}, \mathtt{0x25}]`, and the ``"textest"`` HRP is used when
 reencoding in step 3.
 
-Wallets and other Senders sending to a TEX address (as any output) MUST
-ensure that only transparent UTXOs are spent in the creation of a
-transaction.
+A TEX address can be parsed by reversing this encoding, i.e.:
+
+1. Decode the address to a byte sequence using Bech32m [#bip-0350]_, checking
+   that the HRP is ``"tex"`` for a Mainnet TEX Address and ``"textest"`` for a
+   Testnet TEX Address.
+2. If the length of the resulting byte sequence is not 20 bytes, return an error.
+   Otherwise, the **validating key hash** is this byte sequence.
+
+Design considerations for Senders
+---------------------------------
+
+If a user desires to spend shielded funds to a TEX Address, a Sender supporting
+this ZIP MUST create two transactions: one that unshields the funds to an
+ephemeral transparent address, and one that spends from that ephemeral address
+to the destination TEX Address. This does not defeat the intent of the ZIP,
+because it is still possible for a Recipient to return the funds to the Sender
+by sending them back to the ephemeral address.
 
-Reference Implementation (Alternative 1)
-----------------------------------------
+Wallets MUST be able to recognize funds that have been returned in this way
+and spend them if desired. In order for this to be possible without use of
+TEX Addresses increasing the risk of loss of funds, wallets based on ZIP 32
+[#zip-0032]_ SHOULD choose ephemeral addresses in a way that allows the
+corresponding private keys to be recovered from a ZIP 32 master seed.
+
+However, ephemeral addresses SHOULD NOT be chosen in a way that allows them
+to be linked between transactions, without knowledge of the wallet seed or
+the relevant transparent viewing keys.
+
+In order to show accurate transaction history to a user, wallets SHOULD
+remember when a particular transaction output was sent to a TEX Address, so
+that they can show that form rather than its P2PKH form. It is acceptable that
+this information may be lost on recovery from seed.
```

Comment:
This isn't the place for it, but a standard for how to format a (potentially-zero-valued) internal shielded change memo in the deshielding transaction could help mitigate this toe-stub potential.